docs/proposal-rose-suite-run.md

```diff
@@ -39,66 +39,237 @@ The remaining points can happen any time after 1) is complete although note that
+  * Do this automatically if no existing `log` directory is found.
+* Note that `cylc play` will always load the latest workflow definition found in
+  the run directory and will respect any command line options (which may alter
+  the workflow definition). Therefore, continuing a workflow using `cylc play`
+  effectively implies a reload.
+  * The processed `cylc.flow` files will be kept so there will be a record of
+    any changes
+    (see also [cylc-flow #3763](https://github.com/cylc/cylc-flow/issues/3763)).
+* Support Rose optional configuration via the environment variable
+  `ROSE_SUITE_OPT_CONF_KEYS` or the option `--rose-opt-conf-key=KEY`.
+
+`cylc reinstall` - new command to re-install workflows.
+* Very similar to the `cylc install` command except that this applies to a
+  previously installed workflow (i.e. you run this command in the directory of
+  the installed workflow, not in the source directory).
+* The rsync will use `--delete` to ensure that any previously installed files
+  which have been removed from the source directory also get removed from the
+  installed workflow.
+  * Note that this will have the effect of removing any files installed via
+    `rose-suite.conf`. They will get recreated by a subsequent reload or play but
+    could affect a running workflow.
+* Any options specified as part of the original install will be retained unless
+  overridden on the command line.
+* Consider a `--dry-run` option to report what would be changed?
+  * Does this replace `rose suite-cmp-vc`?
+
+`cylc reload` - no change from existing command.
+
+`cylc validate` - as per existing command with the following changes:
+* If validating a workflow installed with `cylc install` then use any relevant
+  command line options specified as part of the install unless overridden on the
+  command line.
+* Will work fine on source directories which include `rose-suite.conf` files.
+* No `--strict` mode
+  (naked dummy tasks no longer allowed - all tasks must be declared).
+  * This one is controversial and requires further discussion!
+* Cyclic graph validation can be quite slow and is currently included as part of
+  `--strict`. Add a `--no-cyclic-graph-validation` option to disable this?
+
+Several combinations of the above commands will be commonly used.
+Propose to support these as separate commands:
+* `cylc install-play` (`cylc ip` for short?)
+* `cylc install-validate`
+* `cylc reinstall-reload`
+
+`cylc pause|unpause` - new commands replacing `cylc hold|release|unhold`
+* Note that when you use `cylc stop` the scheduler will be modified to unpause
+  the workflow before stopping.
+  This means that a stopped workflow will not be paused when it is continued via
+  `cylc play`. On the other, a workflow which is continued for any other reason
+  (e.g. the server died) will retain its paused status.
+
+`cylc register` - remove: functionality no longer required.
```

Comment:
:open_mouth: holy sh@t, that the end of an era innit.

Reply:
TBH we've really just renamed `cylc register` to `cylc install`, the same functionality is there :/

Reply:
Meh, `cylc install` does a lot more!

Reply:
(And "install" is a more intuitive concept too, for users, IMO).